components/email/src/Email.svelte

```diff
@@ -500,6 +504,125 @@
+    const subject = currentMessage.subject?.toLowerCase().startsWith("re:")
+      ? currentMessage.subject
+      : `Re: ${currentMessage.subject}`;
+
+    const value = {
+      reply_to_message_id: currentMessage.id,
+      from: from,
+      to: to,
+      reply_to: from,
+      subject: subject,
+    };
+
+    if (_this.with_composer) {
+      dispatchEvent("replyAllClicked", {
+        event,
+        message: activeThread.messages[msgIndex],
+        thread: activeThread,
+        value,
+        to,
+        from,
+        subject,
+      });
+    }
+  }
+
+  async function handleReplyClick(event: MouseEvent, msgIndex: number) {
+    event.stopImmediatePropagation();
+
+    const currentMessage = activeThread.messages[msgIndex];
+
+    const me: Participant = {
+      name: _this.you.name,
+      email: _this.you.email_address,
+    };
+    const participantsWithoutMe = activeThread.participants.filter(
+      (e) => e.email != me.email,
+    );
+
+    const from = [me];
+    let to;
+    const subject = currentMessage.subject?.toLowerCase().startsWith("re:")
+      ? currentMessage.subject
+      : `Re: ${currentMessage.subject}`;
+
+    /**
+     * In Gmail, reply options are available on each message in thread.
+     * There are a couple cases that need to be handled when the use clicks 'reply'
+     * and there are multiple participants in an email thread. In some cases, participants
+     * are add to the thread after messages have already been exchanged with out them.
+     *
+     *
+     * 1. When the message is from the user, AND the message is to multiple participants of the thread
+     *    then the default action is to reply to all participants
+     * 2. When the message is from the user, AND the message is to a single participant of the thread
+     *    then reply to only that participant
+     * 3. When the message is NOT from the user, then reply to the sender of the message
+     */
+    if (isFromMe(currentMessage)) {
+      if (currentMessage.to.length > 1) {
+        to = participantsWithoutMe;
+      } else {
+        to = currentMessage.to;
+      }
+    } else {
+      to = currentMessage.from;
+    }
+
+    const value = {
+      reply_to_message_id: currentMessage.id,
+      from: from,
+      to: to,
+      reply_to: from,
+      subject: subject,
+    };
+
+    if (_this.with_composer) {
+      dispatchEvent("replyClicked", {
+        event,
+        message: activeThread.messages[msgIndex],
+        thread: activeThread,
+        value,
+        to,
+        from,
+        subject,
+      });
+    }
+  }
+
+  function isFromMe(message: Message): boolean {
+    if (!_this.you.email_address) {
+      return false;
+    }
+
+    return message.from
+      .map((f) => {
+        return f.email.toLowerCase();
+      })
+      .includes(_this.you.email_address?.toLowerCase());
+  }
+
+  function canReplyAll(message: Message): boolean {
+    return !isFromMe(message) && message.to.length > 1;
```

Comment:
nit - switching these conditions around would be slightly more performant since we wouldn't have to run `isFromMe()` if `message.to.length <= 1`